### PR TITLE
refactor(useQueries): export QueriesOptions type

### LINKED ISSUE #3250: Export QueriesOptions
This issue is for exporting the type `QueriesOptions` for `useQueries`

yes, we should export it. Thank you

_Originally posted by @TkDodo in https://github.com/tannerlinsley/react-query/discussions/3242#discussioncomment-2076367_

### PR DESCRIPTION
**Pull request by jmsims2**
export the QueriesOptions type

closes #3250

### CODE REVIEW

**Review by rkulinski — APPROVED** [review]

File: src/react/useQueries.ts [hunk 1]
```diff
@@ -68,7 +68,7 @@ type GetResults<T> =
 /**
  * QueriesOptions reducer recursively unwraps function arguments to infer/enforce type param
  */
-type QueriesOptions<
+export type QueriesOptions<
```
Comment (TkDodo) [inline src/react/useQueries.ts:71]: can we export the `QueriesResults` too please?

Comment (jmsims2) [reply src/react/useQueries.ts:71]: @TkDodo All set!

**Review by TkDodo — APPROVED** [review]

**Pull request merged**